src/haz3lweb/app/helpful-assistant/Assistant.re

```diff
@@ -0,0 +1,1172 @@
+module Sexp = Sexplib.Sexp;
+open Haz3lcore;
+open Util;
+open Util.OptUtil.Syntax;
+open StringUtil;
+
+module CodeModel = CodeEditable.Model;
+
+module Model = {
+  [@deriving (show({with_path: false}), sexp, yojson)]
+  type party =
+    | System
+    | LLM
+    | LS;
+
+  [@deriving (show({with_path: false}), sexp, yojson)]
+  type message = {
+    party,
+    code: option((Segment.t, option(Id.t))),
```

Comment:
this code type isn't completely obvious, especially the nested option. maybe separately alias this and add some brief comments